include/RAJA/policy/desul/atomic.hpp

```diff
@@ -0,0 +1,178 @@
+//~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~//
+// Copyright (c) 2016-21, Lawrence Livermore National Security, LLC
+// and RAJA project contributors. See the RAJA/COPYRIGHT file for details.
+//
+// SPDX-License-Identifier: (BSD-3-Clause)
+//~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~//
+
+#ifndef RAJA_policy_atomic_desul_HPP
+#define RAJA_policy_atomic_desul_HPP
+
+#include "RAJA/config.hpp"
+
+#if defined(RAJA_ENABLE_DESUL_ATOMICS)
+
+#include "RAJA/util/macros.hpp"
+
+#include "RAJA/policy/atomic_builtin.hpp"
+
+#include "desul/atomics.hpp"
+
+// Default desul options for RAJA
+using raja_default_desul_order = desul::MemoryOrderRelaxed;
+using raja_default_desul_scope = desul::MemoryScopeDevice;
+
+
+namespace RAJA
+{
+
+RAJA_SUPPRESS_HD_WARN
+template <typename AtomicPolicy, typename T>
+RAJA_HOST_DEVICE
+RAJA_INLINE T
+atomicAdd(AtomicPolicy, T volatile *acc, T value) {
```

Comment:
@davidbeckingsale I just realized that we could maybe have defaults for  ``AtomicPolicy`` for each function here since it just gets ignored anyways. It would allow for users to say something like ``RAJA::atomicAdd(pointer, value);`` without needing to specify a dummy-policy. This should maintain the backwards compatibility granted by ``AtomicPolicy`` as well.

Reply:
Actually, I'm not sure that would work out in preserving backward compatibility since ``AtomicPolicy`` comes first in the parameter order.

Reply:
Sadly not, defaults can only be at the end.  It would have to be an overload, and the overload would probably be ambiguous with the templated one sadly.